portable/MSVC-MingW/port.c

```diff
@@ -29,12 +29,31 @@
 /* Standard includes. */
 #include <stdio.h>
 
+/** Exclude Windows APIs such as Cryptography, DDE, RPC, Shell,
```

Comment:
I am not clear on why these changes are needed here. Would you please elaborate?

Reply:
Sorry, I had written a short explanation of this in the description, but I should have explained more.

In MbedTLS versions after, and including, v3.5.0 they started defining `WIN32_LEAN_AND_MEAN`, and using `winsock2.h` The commit for this can be found [here](https://github.com/Mbed-TLS/mbedtls/commit/c26bd760204b27149a4bb08bf2604cc34d3a85fd), and the blame can be found [here](https://github.com/Mbed-TLS/mbedtls/blame/v3.5.0/library/x509_crt.c#L2697)

As such in order to upgrade the MbedTLS version from 3.2.1 to 3.5.1 in the FreeRTOS/FreeRTOS repo we need to use winsock2.h in the MSVC port.

Does that explain why this is being done?